neuro_san/internals/run_context/langchain/journaling/journaling_passthrough.py

```diff
@@ -0,0 +1,120 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from typing import Any
+from typing import Dict
+from typing import List
+
+from copy import copy as shallow_copy
+
+from pydantic import ConfigDict
+
+from langchain_core.messages.ai import AIMessage
+from langchain_core.messages.base import BaseMessage
+from langchain_core.messages.tool import ToolMessage
+from langchain_core.runnables.passthrough import RunnablePassthrough
+
+from neuro_san.internals.journals.journal import Journal
+
+
+class JournalingPassthrough(RunnablePassthrough, Journal):
+    """
+    RunnablePassthrough implementation that intercepts journal messages
+    """
```

Comment:
Create a new RunnablePassthrough guy that can intercept journal messages.
This isn't entirely working the way that I want it just yet, but this is a decent checkpoint for future enhancements.